src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,108 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private static final int RESIZE_COEFFICIENT = 2;
+    private Node<K, V>[] table;
+    private int threshold;
+    private int size;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_CAPACITY];
+        threshold = (int) (LOAD_FACTOR * DEFAULT_CAPACITY);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        int index = calculateIndexByKey(key);
+        putValue(index, key, value);
     }
 
     @Override
     public V getValue(K key) {
+        int index = calculateIndexByKey(key);
+        Node<K, V> node = table[index];
+
+        while (node != null) {
+            K currentKey = node.key;
+            if (compareKeys(currentKey, key)) {
+                return node.value;
+            }
+            node = node.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int getHashForKey(K key) {
+        return (key == null) ? 0 : Math.abs(key.hashCode());
+    }
+
+    private int calculateIndexByKey(K key) {
+        return getHashForKey(key) % table.length;
+    }
+
+    private void putValue(int index, K key, V value) {
+        if (size + 1 > threshold) {
+            resize();
+        }
+        Node<K, V> node = table[index];
+        Node<K, V> newNode = new Node<>(key, value, null);
+
+        while (node != null) {
+            K currentKey = node.key;
+            if (compareKeys(currentKey, key)) {
+                node.value = value;
+                return;
+            } else if (node.next == null) {
```

Comment:
```suggestion
            } if (node.next == null) {
```
we don't need else after return